pkg/log/text_formatter.go

```diff
@@ -15,191 +15,67 @@
 package log
 
 import (
+	"bufio"
 	"bytes"
-	"fmt"
-	"runtime"
-	"sort"
 	"strings"
-	"time"
 
 	"github.com/Sirupsen/logrus"
 )
 
-const (
-	nocolor = 0
-	red     = 31
-	green   = 32
-	yellow  = 33
-	blue    = 34
-	gray    = 37
-)
-
-var (
-	baseTimestamp time.Time
-	isTerminal    bool
-)
-
-func init() {
-	baseTimestamp = time.Now()
-	isTerminal = logrus.IsTerminal()
```

Comment:
Have you tested this with and without a `tty` attached?  IIRC if you redirect stdout to a file vs let stdout print to console, you got different formatting behavior.  Basically `\n` and control characters would end up in your log files.